File: src/main/graphql/io/github/graphql/model/CreateRefMutationResponse.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult

@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-05T19:09:00+0800"
)
open class CreateRefMutationResponse : GraphQLResult<MutableMap<String, CreateRefPayloadTO?>>() {

    companion object {
        const val OPERATION_NAME: String = "createRef"
    }

    fun createRef(): CreateRefPayloadTO? {
        val data: MutableMap<String, CreateRefPayloadTO?> = super.getData()
        return data[OPERATION_NAME]
    }
}

File: src/main/scala-graphql/io/github/graphql/scala/model/CreateRefMutationResponse.scala
package io.github.graphql.scala.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult
import java.util.{ Map => JMap }

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-07T19:49:37+0800"
)
class CreateRefMutationResponse extends GraphQLResult[JMap[String, CreateRefPayloadTO]] {

    def createRef(): CreateRefPayloadTO = {
        val data: JMap[String, CreateRefPayloadTO] = getData
        if (data != null) data.get(CreateRefMutationResponse.OPERATION_NAME) else null
    }

}

object CreateRefMutationResponse {

    private final val OPERATION_NAME: String = "createRef"

}
